### DISCUSSION TITLE
v1.3.0

### DISCUSSION THREAD

**Discussion by marcus-crane** [marcus-crane/october#68] [Announcements]
This release refactors all of the internal logging, dropping [zerolog](https://github.com/rs/zerolog) in favour of trusty [logrus](https://github.com/Sirupsen/logrus) which is not user facing at all.

As a result, logging now works properly again on Windows closing #25

While going through all the various log entries, I've also switched the logging format to JSON which makes it easier to parse through each file. Some log entry sizes have also been greatly reduced where before they would log out an entire struct even if the whole thing wasn't useful. This should greatly reduce the size of the average log file.

In order to making it easier for users to submit logs for investigations, there is now a button in the Settings view that will open your computer's file explorer to the location that October stores logs in (#12)

**Windows**
![windows-logs](https://user-images.githubusercontent.com/14816406/206837799-237f9dbd-74eb-4530-9e79-4e45b87059e8.png)

**macOS**
<img width="1410" alt="CleanShot 2022-12-10 at 20 21 20@2x" src="https://user-images.githubusercontent.com/14816406/206837886-755d068d-a505-4ae6-a5dc-8d948e15f953.png">

You might also notice a couple of other small additions in the screenshots above.

There is now a little build identifier in the Settings view where you can see build information about the version of October you're running (#65) as well as a button for reporting bugs which will open a new Github issue pre-populated with said build information to make it easier to provide support to end users.

I've still got some other refactoring tasks to do which should hopefully then make it easier to add in some of the larger features that have been outstanding such as better handling of upload failures as well as cutting down the time to upload

**Comment by kraftskunk** [comment]
AM I the only one to have a virus issue when unpacking the *.zip?

Norton is guaranteeing the *.exe because it identieifes it as a virus.

**Reply by marcus-crane** [reply]
Hi there,

While I can't speak to any particular antivirus software, there was a bit of discussion around this earlier that you can read [in this thread](https://github.com/marcus-crane/october/discussions/39#discussioncomment-2555562).

As mentioned, usually October builds ends up being uploaded to VirusTotal or one of the other bulk antivirus scanners and one of them generally always triggers a false positive.

The particular problem is to do with something called WebView, which is a base part of October. It's essentially an embedded browser, not unlike [Electron](https://www.electronjs.org/), but much more lightweight and built into the base operating system.

For macOS and Linux users, this is built directly into the operating system while for Windows users, this is downloadable as [Microsoft Edge WebView 2](https://learn.microsoft.com/en-us/microsoft-edge/webview2/). I believe it's installed by default with Windows 11 but has to be downloaded in earlier Windows versions.

Since it isn't guaranteed to be installed for Windows users, October embeds the installer so that if WebView is missing, you can actually use the software.

Now, you could of course install it yourself which is slightly less convenient but perhaps it would be better in the long run because it appears that a small handful of antivirus scanners detected the WebView installer embedded within October and claim that it is a piece of malware.

Looking [at the latest VirusTotal scan](https://www.virustotal.com/gui/file/64b6f704e66769e8b208f30be95f90c1c578f83f54014f4aeed9d70c259252a8/community), it appears that one rule for one antivirus vendor matches whenever a Microsoft Copyright string is embedded in a piece of non-Microsoft software.

As for the fix, it can likely be resolved if I switch from bundling the WebView installer with October to installed just redirecting the user to install it themselves. It's slightly more work for setup but it should reduce the download size a little and make no difference for Windows 11+ users.

Anyway, you don't have to trust the prebuilt versions that I've precompiled as the source code is freely available.

You're more than welcome to download the source code, audit it back to front and then build the project from scratch on your local machine if you want to be 100% sure for yourself.

While I don't use Windows much myself, the only antivirus software that I run October through is Microsoft Smartscreen as the developer certificate attached to the October builds is influenced by the results of Smartscreen. If Microsoft decided one day that October was malware, all of the existing builds would throw up a warning as the reputation of the developer certificate would be flagged affecting everything it signs.

As far as resolving false positives, I'll probably switch over to prompting Windows users to install WebView themselves (removing the embedded installer) as I can't really do anything to address the results of the handful of antivirus scanners that pop up so if they're triggering on the embed, I'll just have to remove it 🙂
